docs/recipes/ui/add-routing-with-separate-states.md

```diff
@@ -0,0 +1,292 @@
+
+also open `Shared`, `Fable.Remoting.Client`, `Elmish` `Feliz` and `Feliz.Bulma`
+
+```fsharp title="TodoList.fs"
+module TodoList
+
+open Shared
+open Fable.Remoting.Client
+open Elmish
+open Feliz
+open Feliz.Bulma
+
+
+type Model = { Todos: Todo list; Input: string }
+
+type Msg =
+    | GotTodos of Todo list
+    | SetInput of string
+    | AddTodo
+    | AddedTodo of Todo
+
+let todosApi =
+    Remoting.createApi ()
+    |> Remoting.withRouteBuilder Route.builder
+    |> Remoting.buildProxy<ITodosApi>
+
+let init () : Model * Cmd<Msg> =
+    let model = { Todos = []; Input = "" }
+    let cmd = Cmd.OfAsync.perform todosApi.getTodos () GotTodos
+
+    model, cmd
+
+let update (msg: Msg) (model: Model) : Model * Cmd<Msg> =
+    match msg with
+    | GotTodos todos -> { model with Todos = todos }, Cmd.none
+    | SetInput value -> { model with Input = value }, Cmd.none
+    | AddTodo ->
+        let todo = Todo.create model.Input
+
+        let cmd = Cmd.OfAsync.perform todosApi.addTodo todo AddedTodo
+
+        { model with Input = "" }, cmd
+    | AddedTodo todo -> { model with Todos = model.Todos @ [ todo ] }, Cmd.none
+
+let view (model: Model) (dispatch: Msg -> unit) =
+    Bulma.box [
+        Bulma.content [
+            Html.ol [
+                for todo in model.Todos do
+                    Html.li [ prop.text todo.Description ]
+            ]
+        ]
+        Bulma.field.div [
+            field.isGrouped
+            prop.children [
+                Bulma.control.p [
+                    control.isExpanded
+                    prop.children [
+                        Bulma.input.text [
+                            prop.value model.Input
+                            prop.placeholder "What needs to be done?"
+                            prop.onChange (fun x -> SetInput x |> dispatch)
+                        ]
+                    ]
+                ]
+                Bulma.control.p [
+                    Bulma.button.a [
+                        color.isPrimary
+                        prop.disabled (Todo.isValid model.Input |> not)
+                        prop.onClick (fun _ -> dispatch AddTodo)
+                        prop.text "Add"
+                    ]
+                ]
+            ]
+        ]
+    ]
+```
+
+## 3. Adding a new Model to the Index page
+
+Create a new Model in the `Index` module, to keep track of the open page
+
+```fsharp title="Index.fs"
+type Page =
+    | TodoList of TodoList.Model
+    | NotFound 
+
+type Model = { CurrentPage: Page }
+```
+
+## 4. Updating the TodoList model
+
+Add a `Msg` type with a case of `TodoList.Msg`
+
+```fsharp title="Index.fs"
+type Msg =
+    | TodolistMsg of TodoList.Msg
+```
+
+Create an `update` function (we moved the original one to `TodoList`). Handle the `TodoListMsg` by updating the `TodoList` Model. Wrap the command returned by the `update` of the todo list in a `TodoListMsg` before returning it
+
+```fsharp title="Index.fs"
```

Comment:
I was expecting to handle the other cases in this section, but I see that they're handled in later sections. I think that perhaps it would be worth a sentence mentioning that other cases will be addressed later.